### ISSUE TITLE
Images should not be indexed on alt, but rather unique id

### ISSUE THREAD

**Issue by stanbar** [ccrsxx/twitter-clone#42]
Hi,
I believe that storing images based on the `alt` is not an good idea and instead unique `id` should be used. The reason is that, uploading two images with the same filename (thus the same `alt`s), replaces the previous image.

Instead of 
https://github.com/ccrsxx/twitter-clone/blob/c6bf6fba409cbff2ae57bc8069f1e00a202853a0/src/lib/firebase/utils.ts#L139

I would rather use

```ts
const storageRef = ref(storage, `images/${userId}/${id}`);
```

**Comment by Ketchupchh** [comment]
Yeah, this should be fixed. In my testing it doesn't replace any images but just discards the new one and places the original in the tweet instead. I'll fix the issues you found in my PR as well, thank you. I'll contact ccrsxx about the username bug as well.

**Comment by stanbar** [comment]
@Ketchupchh thank you!

**Comment by ccrsxx** [comment]
Yup, but that's by design.

I made it so that when we upload the same image with the same alt, it will return the old image instead of creating the new image. It won't replace the old image with the new image tho. 

This is the code:
https://github.com/ccrsxx/twitter-clone/blob/c6bf6fba409cbff2ae57bc8069f1e00a202853a0/src/lib/firebase/utils.ts#L141-L146

I did it to reduce the bandwidth usage on the storage. 

But yeah, that's bad if you upload a different picture with the same alt, it'll return the old one.

I see that the bandwidth usage isn't that much, so I think replacing the alt to id is good idea now.

**Comment by stanbar** [comment]
@ccrsxx Well, if you want to do this kind of optimisation I think that the better option would be to calculate a hash (e.g. sha256) of a file's content, and use that instead of a filename. That way, the same images with different names, would be stored only once.

**Comment by ccrsxx** [comment]
Yup, that could be the solution. But for now, I''ll be using the randomly generated id.

**Comment by ccrsxx** [comment]
Fixed in #40
